addon_service/common/serializer_fields.py

```diff
@@ -48,3 +49,19 @@ def __init__(self, dataclass_model):
 
     def get(self, *, pk):
         return self.model.get_by_pk(pk)
+
+
+class CustomPolymorphicResourceRelatedField(PolymorphicResourceRelatedField):
+    def to_representation(self, value):
+        data = super().to_representation(value)
+        if hasattr(value, "authorizedcitationaccount"):
+            data["type"] = "authorized-citation-account"
+        elif hasattr(value, "authorizedstorageaccount"):
+            data["type"] = "authorized-storage-account"
+        elif hasattr(value, "configuredstorageaddon"):
+            data["type"] = "configured-storage-addon"
+        elif hasattr(value, "configuredstorageaccount"):
+            data["type"] = "configured-storage-account"
```

Comment:
@felliott This will be relevant to you once this is merged.

Reply:
I noticed that there were wrong checks and types, so I fixed them in a new commit

Reply:
thanks for the heads up!